scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkWriter.scala

```diff
@@ -165,7 +177,7 @@ class SparkWriter(val sparkMode: SparkMode) extends Writer {
     }
     def write[T](opt: TypedPipe[T], keyPipe: TypedPipe[T], sink: TypedSink[T], oldState: State): (State, Action) = {
       val promise = Promise[RDD[_ <: T]]()
-      val (newState, added) = oldState.addForce[T](conf, keyPipe, opt, promise.future)
+      val (newState, added) = oldState.addForce[T](conf, init = keyPipe, opt = opt, promise.future, None)
```

Comment:
there is no need to default force to disk first, if the user wanted that, they should have used .forceToDisk or forceToDiskExecution.